joint_state_broadcaster/src/joint_state_broadcaster.cpp

```diff
@@ -346,18 +403,19 @@ double get_value(
 controller_interface::return_type JointStateBroadcaster::update(
   const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
 {
-  for (const auto & state_interface : state_interfaces_)
+  for (auto i = 0u; i < state_interfaces_.size(); ++i)
   {
-    std::string interface_name = state_interface.get_interface_name();
-    if (map_interface_to_joint_state_.count(interface_name) > 0)
+    /// @note should we limit the number of retries here? does it even
+    /// make sense to try to get the latest value on every iteration?
+    const auto & opt = state_interfaces_[i].get_optional();
```

Comment:
```suggestion
    const auto & opt = state_interfaces_[i].get_optional(0);
```
Can you set the number of retries to zero (or) one?

Reply:
Done,